main.go

```diff
@@ -111,20 +112,31 @@ func (c *Controller) handleNewCSR(key string) error {
 		return nil
 	}
 
-	if pending := recentlyPendingCSRs(c.indexer); pending > maxPendingCSRs {
-		klog.Infof("ignoring all CSRs as too many recent pending CSRs seen: %d", pending)
-		return nil
+	machines, err := c.machines.List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list machines: %v", err)
 	}
 
-	parsedCSR, err := parseCSR(csr)
+	csrsList, err := c.csrs.List(metav1.ListOptions{})
 	if err != nil {
-		klog.Infof("error parsing request CSR: %v", err)
+		return fmt.Errorf("failed to list all csrs: %v", err)
+	}
+
+	maxPending := maxPendingTotal(csrsList.Items, machines.Items, maxPendingOver)
+	if err != nil {
+		return err
+	}
+
+	if pending := recentlyPendingCSRs(c.indexer); pending > maxPending {
+		klog.Error("Difference between pending CSRs and machines > 100", pending)
+		klog.Errorf("Pending CSRs: %d; Max pending allowed: %d", pending, maxPending)
 		return nil
```

Comment:
I think it was a mistake for me to `return nil` here in the original implementation - let us return an error to force a few retries.  That way if the indexer cache is lagging behind, we do not give up too early.